src/config.py

```diff
@@ -1,82 +1,76 @@
 import os
-from typing import Type, TypeVar
+from typing import Optional
 
 from dotenv import load_dotenv
 
-# Generic type, used for static type checking
-T = TypeVar("T", str, bool, int, float)
-
 load_dotenv()
 load_dotenv("api_keys.env")
 
 
-def get_env_variable(var_name: str,
-                     default: T = None, allow_empty: bool = False, cast_to: Type[T] = str) -> T:
+def get_env_variable(var_name: str, default: Optional[str] = None, allow_empty: bool = False) -> str:
```

Comment:
I think I would need to talk to you to get a good grasp on some of the confusions in these conversations, so that we can come up with a good solution. I have a couple ideas on how it can be improved but they might clash with your idea

Reply:
Let me know if you think my new class `EnvVariable` resolves the complexity enough